File: highspeed_report/highspeed_accounts/report/simple_sales_log/simple_sales_log.js
// Copyright (c) 2025, Your Company and contributors
// For license information, please see license.txt

frappe.query_reports["Simple Sales Log"] = {
    "filters": [
        {
            "fieldname": "company",
            "label": __("Company"),
            "fieldtype": "Link",
            "options": "Company",
            "default": frappe.defaults.get_user_default("Company"),
            "reqd": 1,
            "width": "150px"
        },
        {
            "fieldname": "from_date",
            "label": __("من تاريخ"),
            "fieldtype": "Date",
            "default": frappe.datetime.add_months(frappe.datetime.get_today(), -1),
            "reqd": 1,
            "width": "100px" 
        },
        {
            "fieldname": "to_date",
            "label": __("إلى تاريخ"),
            "fieldtype": "Date",
            "default": frappe.datetime.get_today(),
            "reqd": 1,
            "width": "100px"
        },
        {
            "fieldname": "from_time",
            "label": __("من وقت"),
            "fieldtype": "Time",
            "default": "00:00:00",
            "reqd": 1,
            "width": "100px"
        },
        {
            "fieldname": "to_time",
            "label": __("إلى وقت"),
            "fieldtype": "Time",
            "default": "23:59:59",
            "reqd": 1,
            "width": "100px"
        },
        {
            "fieldname": "customer",
            "label": __("Customer"),
            "fieldtype": "Link",
            "options": "Customer",
            "width": "150px",
            "get_query": function() {
                return {
                    "filters": [
                        ["Customer", "disabled", "=", 0]
                    ]
                };
            }
        },
        {
            "fieldname": "status",
            "label": __("حالة الفاتورة"),
            "fieldtype": "Select",
            "options": "\nPaid\nUnpaid\nPartly Paid\nOverdue\nCredit Note Issued\nReturn",
            "width": "150px",
            "translatedOptions": {
                "Paid": __("مسددة بالكامل"),
                "Unpaid": __("غير مسددة"),
                "Partly Paid": __("مسددة جزئياً"),
                "Overdue": __("متأخرة السداد"),
                "Credit Note Issued": __("إشعار دائن مصدر"),
                "Return": __("مرتجع")
            }
        },
        {
            "fieldname": "invoice_type",
            "label": __("نوع الفاتورة"),
            "fieldtype": "Select",
            "options": "\nSales\nReturn",
            "default": "",
            "width": "100px",
            "translatedOptions": {
                "Sales": __("مبيعات"),
                "Return": __("مرتجعات")
            }
        }
    ],
    "formatter": function(value, row, column, data, default_formatter) {
        if (data) {
            // Avoid showing "null" in any column
            if (value === null || value === "null") {
                value = "";
            }
            
            // Format invoice status - with color coding
            if (column.fieldname === "invoice_status") {
                if (!value) return '';
                
                let color = "black";
                if (value.includes("مرتجع")) {
                    color = "red";
                } else if (value.includes("مسددة بالكامل")) {
                    color = "green";
                } else if (value.includes("غير مسددة") || value.includes("متأخرة")) {
                    color = "orange";
                } else if (value.includes("مسددة جزئياً")) {
                    color = "blue";
                } else if (value.includes("ملغية")) {
                    color = "gray";
                }
                
                return `<span style="color: ${color}; font-weight: bold;">${value}</span>`;
            }
            
            // Format for amount column
            if (column.fieldname === "grand_total") {
                return '<span style="font-weight: bold;">' + default_formatter(value, row, column, data) + '</span>';
            }
            
            // Format for voucher_no column - add link
            if (column.fieldname === "voucher_no") {
                return `<a href="/app/sales-invoice/${value}" target="_blank">${value}</a>`;
            }
        }
        
        return default_formatter(value || "", row, column, data);
    },

    "onload": function(report) {
        console.log("تم تحميل تقرير سجل المبيعات البسيط");
        
        try {
            // Add custom CSS
            $('<style>\
                .datatable .dt-cell { padding: 3px 6px !important; }\
                .datatable .dt-row { border-bottom: 1px solid #eee; }\
                .datatable .dt-header .dt-cell { background-color: #f5f7fa; font-weight: bold; }\
                .sales-summary-section { direction: rtl; }\
                .summary-card { background-color: #f8f8f8; border-radius: 5px; padding: 15px; margin-bottom: 10px; box-shadow: 0 2px 5px rgba(0,0,0,0.1); }\
                .summary-card-title { font-weight: bold; margin-bottom: 10px; font-size: 16px; }\
                .summary-amount { font-size: 20px; font-weight: bold; }\
                .summary-card.total { border-right: 5px solid #4272d7; }\
                .summary-card.invoice-count { border-right: 5px solid #28a745; }\
                .total-amount { color: #4272d7; }\
                .invoice-count-value { color: #28a745; }\
            </style>').appendTo('head');
            
            // Create sales summary section
            var $salesSummarySection = $('<div class="sales-summary-section" style="margin-bottom: 20px; padding: 15px; background-color: #fff; border: 1px solid #ddd; border-radius: 5px; box-shadow: 0 2px 5px rgba(0,0,0,0.05);"></div>');
            var $salesSummaryHeader = $('<div style="display: flex; justify-content: space-between; align-items: center; margin-bottom: 15px; border-bottom: 1px solid #f0f0f0; padding-bottom: 10px;"><h3 style="margin: 0; color: #333; font-weight: bold;">' + __("ملخص سجل المبيعات") + '</h3><div class="report-date"></div></div>');
            
            // Create row for summary cards
            var $summaryCardsSection = $('<div class="summary-cards" style="display: flex; flex-wrap: wrap; gap: 15px; margin-bottom: 15px;"></div>');
            
            // Create summary cards
            var $totalCard = $('<div class="summary-card total" style="flex: 1;"><div class="summary-card-title">' + __("إجمالي المبيعات") + '</div><div class="summary-amount total-amount"></div></div>');
            var $invoiceCountCard = $('<div class="summary-card invoice-count" style="flex: 1;"><div class="summary-card-title">' + __("عدد الفواتير") + '</div><div class="summary-amount invoice-count-value"></div></div>');
            
            $summaryCardsSection.append($totalCard).append($invoiceCountCard);
            $salesSummarySection.append($salesSummaryHeader).append($summaryCardsSection);
            
            // Add section to the top of the report after filters
            if (report.page.main.find('.sales-summary-section').length === 0) {
                report.page.main.find('.report-filter-section').after($salesSummarySection);
            }
            
            // Update sales summary when data is loaded
            var originalRefresh = report.refresh;
            report.refresh = function() {
                originalRefresh.call(this);
                
                // Check if data is available
                if (report.data && report.data.length > 0) {
                    try {
                        updateSalesSummary(report.data, report.get_values());
                    } catch (e) {
                        console.error("Error updating sales summary:", e);
                        frappe.msgprint({
                            title: __("Error"),
                            message: __("حدث خطأ أثناء تحديث ملخص المبيعات. يرجى تحديث الصفحة."),
                            indicator: 'red'
                        });
                    }
                }
            };
            
            // Add print button
            report.page.add_inner_button(__("Print Report"), function() {
                if (report.data && report.data.length > 0) {
                    printReport(report.data, report.get_values());
                } else {
                    frappe.msgprint(__("لا توجد بيانات لعرضها. يرجى التحقق من المرشحات."));
                }
            });
            
            // Add export button
            report.page.add_inner_button(__("تصدير"), function() {
                if (report.data && report.data.length > 0) {
                    downloadCSV(report.data, "simple_sales_log_" + frappe.datetime.get_today() + ".csv");
                } else {
                    frappe.msgprint(__("لا توجد بيانات للتصدير"));
                }
            });
        
        } catch (e) {
            console.error("Error in Simple Sales Log onload:", e);
            frappe.msgprint({
                title: __("Error"),
                message: __("حدث خطأ أثناء تهيئة التقرير. يرجى تحديث الصفحة."),
                indicator: 'red'
            });
        }
    }
};

// Function to update sales summary
function updateSalesSummary(data, filters) {
    // Update report date
    $(".report-date").html(`<span style="font-size: 14px;">${__("تاريخ التقرير")}: ${formatDate(frappe.datetime.get_today())} ${formatTime(frappe.datetime.now_time())}</span>`);
    
    // Separate regular rows from total rows
    var regularRows = [];
    var totalRow = null;
    
    // Identify regular rows and the total row (if any)
    data.forEach(function(row) {
        if (row.is_total_row) {
            totalRow = row;
        } else {
            regularRows.push(row);
        }
    });
    
    // Calculate totals properly from regular rows only
    var totalAmount = regularRows.reduce((sum, row) => sum + flt(row.grand_total), 0);
    var invoiceCount = regularRows.length;
    
    // Update summary cards with the correct values
    $(".total-amount").text(format_currency(totalAmount));
    $(".invoice-count-value").text(invoiceCount);
}

// Function to print the report
function printReport(data, filters) {
    try {
        // Show wait message while preparing the report
        frappe.show_alert({
            message: __("جاري تجهيز التقرير للطباعة..."),
            indicator: 'blue'
        });
    
        // Get company information
        frappe.db.get_value('Company', filters.company, ['company_name', 'tax_id'], function(companyInfo) {
            // Create sales table rows
            var salesRows = [];
            
            // We will only use rows that contain actual invoice data
            // This means excluding ANY row that might be a total, subtotal, or summary row
            var actualDataRows = [];
            
            // Identify actual invoice rows - rows with a valid voucher_no are actual data
            data.forEach(function(row) {
                // Only include rows that have an actual invoice number and are not summary/total rows
                if (row.voucher_no && !row.is_total_row && row.grand_total !== undefined) {
                    actualDataRows.push(row);
                }
            });
            
            // Calculate correct total from actual data rows only
            var totalAmount = actualDataRows.reduce((sum, row) => sum + flt(row.grand_total), 0);
            
            // Add only the actual data rows to the output
            actualDataRows.forEach(function(row) {
                salesRows.push(`
                    <tr>
                        <td>${formatDate(row.posting_date)} ${formatTime(row.posting_time)}</td>
                        <td>${row.voucher_no}</td>
                        <td>${row.voucher_type}</td>
                        <td>${row.customer_name}</td>
                        <td>${row.invoice_status}</td>
                        <td>${format_currency(row.grand_total)}</td>
                        <td>${row.owner}</td>
                    </tr>
                `);
            });
            
            // Add only one total row with the correct total
            salesRows.push(`
                <tr style="font-weight: bold; background-color: #f2f2f2;">
                    <td colspan="5" style="text-align: center;">${__("Grand Total")}</td>
                    <td>${format_currency(totalAmount)}</td>
                    <td></td>
                </tr>
            `);
            
            // Build HTML template for printing
            var currentDate = frappe.datetime.get_today();
            var currentTime = frappe.datetime.now_time();
            
            var html = `
            <!DOCTYPE html>
            <html dir="rtl">
            <head>
                <meta charset="UTF-8">
                <title>سجل المبيعات</title>
                <style>
                    @page {
                        size: A4;
                        margin: 1cm;
                    }
                    body { 
                        font-family: Arial, sans-serif; 
                        margin: 0; 
                        padding: 0;
                        direction: rtl;
                        font-size: 12px;
                    }
                    .company-info {
                        text-align: center;
                        margin-bottom: 15px;
                        padding: 5px;
                        border-bottom: 1px solid #000;
                    }
                    .company-info h3 {
                        margin: 5px 0;
                        font-size: 18px;
                    }
                    table { 
                        width: 100%; 
                        border-collapse: collapse; 
                        margin-bottom: 20px; 
                    }
                    th, td { 
                        border: 1px solid #000; 
                        padding: 5px; 
                        text-align: center;
                    }
                    th { 
                        background-color: #f2f2f2; 
                        font-weight: bold;
                    }
                    .report-title {
                        text-align: center;
                        margin: 10px 0;
                        font-size: 16px;
                        font-weight: bold;
                    }
                    .footer {
                        text-align: center;
                        margin-top: 20px;
                        font-size: 10px;
                        border-top: 1px solid #000;
                        padding-top: 5px;
                    }
                    @media print {
                        body { margin: 0; }
                    }
                </style>
            </head>
            <body>
                <div class="company-info">
                    <h3>${companyInfo.company_name}</h3>
                    <div>الرقم الضريبي: ${companyInfo.tax_id || "---"}</div>
                </div>
                
                <div class="report-title">
                    سجل المبيعات / Sales Log
                </div>
                
                <table class="info-table">
                    <tr>
                        <td style="width: 20%; background-color: #f2f2f2; font-weight: bold;">الفترة / Period:</td>
                        <td style="width: 30%;">${formatDate(filters.from_date)} ${formatTime(filters.from_time)} - ${formatDate(filters.to_date)} ${formatTime(filters.to_time)}</td>
                        <td style="width: 20%; background-color: #f2f2f2; font-weight: bold;">تاريخ التقرير / Report Date:</td>
                        <td style="width: 30%;">${formatDate(currentDate)} ${formatTime(currentTime)}</td>
                    </tr>
                    ${filters.customer ? `
                    <tr>
                        <td style="background-color: #f2f2f2; font-weight: bold;">العميل / Customer:</td>
                        <td colspan="3">${filters.customer}</td>
                    </tr>` : ''}
                    ${filters.status ? `
                    <tr>
                        <td style="background-color: #f2f2f2; font-weight: bold;">حالة الفاتورة / Status:</td>
                        <td colspan="3">${filters.status}</td>
                    </tr>` : ''}
                    ${filters.invoice_type ? `
                    <tr>
                        <td style="background-color: #f2f2f2; font-weight: bold;">نوع الفاتورة / Type:</td>
                        <td colspan="3">${filters.invoice_type === 'Sales' ? 'مبيعات' : filters.invoice_type === 'Return' ? 'مرتجعات' : filters.invoice_type}</td>
                    </tr>` : ''}
                </table>
                
                <table class="sales-table">
                    <thead>
                        <tr>
                            <th>التاريخ والوقت<br/>Date & Time</th>
                            <th>رقم المستند<br/>Document No</th>
                            <th>نوع المستند<br/>Type</th>
                            <th>العميل<br/>Customer</th>
                            <th>حالة الفاتورة<br/>Status</th>
                            <th>المبلغ<br/>Amount</th>
                            <th>المستخدم<br/>User</th>
                        </tr>
                    </thead>
                    <tbody>
                        ${salesRows.join('')}
                    </tbody>
                </table>
                
                <div class="footer">
                    تم إصدار هذا التقرير بتاريخ ${formatDate(currentDate)} ${formatTime(currentTime)}
                </div>
            </body>
            </html>
            `;
            
            // Open print window
            var w = window.open();
            if (w) {
                w.document.write(html);
                w.document.close();
                setTimeout(function() {
                    w.print();
                }, 1000);
            } else {
                frappe.msgprint(__("تم حظر النوافذ المنبثقة. يرجى السماح بالنوافذ المنبثقة لطباعة التقرير."));
            }
        }).catch(function(error) {
            console.error("Error getting company info:", error);
            frappe.msgprint({
                title: __("Error"),
                message: __("حدث خطأ أثناء جلب معلومات الشركة للطباعة."),
                indicator: 'red'
            });
        });
    } catch (e) {
        console.error("Error in print report function:", e);
        frappe.msgprint({
            title: __("Error"),
            message: __("حدث خطأ أثناء تجهيز التقرير للطباعة."),
            indicator: 'red'
        });
    }
}

// Function to export data as CSV
function downloadCSV(data, filename) {
    try {
        var csvContent = "data:text/csv;charset=utf-8,\ufeff"; // Add BOM for Arabic support
        
        // Add report title and filters
        csvContent += "سجل المبيعات - Simple Sales Log\r\n";
        const filters = frappe.query_report.get_values();
        csvContent += "الفترة / Period:," + formatDate(filters.from_date) + " " + formatTime(filters.from_time) + " - " + formatDate(filters.to_date) + " " + formatTime(filters.to_time) + "\r\n";
        
        if (filters.customer) {
            csvContent += "العميل / Customer:," + filters.customer + "\r\n";
        }
        
        if (filters.status) {
            csvContent += "حالة الفاتورة / Status:," + filters.status + "\r\n";
        }
        
        if (filters.invoice_type) {
            let typeText = filters.invoice_type === 'Sales' ? 'مبيعات' : 
                           filters.invoice_type === 'Return' ? 'مرتجعات' : filters.invoice_type;
            csvContent += "نوع الفاتورة / Type:," + typeText + "\r\n";
        }
        
        csvContent += "\r\n"; // Add blank line after filters
        
        // Prepare headers
        var headers = ["التاريخ", "الوقت", "رقم المستند", "نوع المستند", __("Customer"), "حالة الفاتورة", __("Amount"), "المستخدم"];
        csvContent += headers.join(",") + "\r\n";
        
        // Separate regular rows from total rows
        var regularRows = [];
        var totalRow = null;
        
        // Identify regular rows and the total row (if any)
        data.forEach(function(row) {
            if (row.is_total_row) {
                totalRow = row;
            } else {
                regularRows.push(row);
            }
        });
        
        // Calculate correct total
        var totalAmount = regularRows.reduce((sum, row) => sum + flt(row.grand_total), 0);
        
        // Add data (only regular rows, not total rows)
        regularRows.forEach(function(row) {
            var rowData = [
                row.posting_date || "",
                row.posting_time || "",
                row.voucher_no || "",
                row.voucher_type || "",
                `"${row.customer_name || ""}"`,
                `"${row.invoice_status || ""}"`,
                row.grand_total || 0,
                `"${row.owner || ""}"`
            ];
            
            csvContent += rowData.join(",") + "\r\n";
        });
        
        // Add total row with the correct total
        csvContent += `"","","","","${__("الإجمالي / Total")}",${totalAmount},""\r\n`;
        
        // Export file
        var encodedUri = encodeURI(csvContent);
        var link = document.createElement("a");
        link.setAttribute("href", encodedUri);
        link.setAttribute("download", filename);
        document.body.appendChild(link);
        link.click();
        document.body.removeChild(link);
    } catch (e) {
        console.error("Error exporting CSV:", e);
        frappe.msgprint({
            title: __("Error"),
            message: __("حدث خطأ أثناء تصدير البيانات."),
            indicator: 'red'
        });
    }
}

// Date formatting function
function formatDate(dateStr) {
    if (!dateStr) return "";
    var d = new Date(dateStr);
    var day = d.getDate().toString().padStart(2, '0');
    var month = (d.getMonth() + 1).toString().padStart(2, '0');
    var year = d.getFullYear();
    return day + '/' + month + '/' + year;
}

// Time formatting function
function formatTime(timeStr) {
    if (!timeStr) return "";
    return timeStr.substr(0, 5); // Show only hours and minutes (HH:MM)
}

File: highspeed_report/highspeed_accounts/report/simple_sales_log/simple_sales_log.py
# Copyright (c) 2025, Your Company and contributors
# For license information, please see license.txt

import frappe
from frappe import _
import datetime

def execute(filters=None):
    """Main execution function for the simplified sales report"""
    if not filters:
        filters = {}
    
    # Set default date range if not provided
    if not filters.get("from_date"):
        filters["from_date"] = datetime.date.today().replace(day=1)
    
    if not filters.get("to_date"):
        filters["to_date"] = datetime.date.today()
    
    # Set default time range if not provided
    if not filters.get("from_time"):
        filters["from_time"] = "00:00:00"
    
    if not filters.get("to_time"):
        filters["to_time"] = "23:59:59"
    
    # Get report data
    columns = get_columns()
    data = get_sales_data(filters)
    
    return columns, data

def get_columns():
    """Define the columns for the simplified report"""
    return [
        {
            "fieldname": "posting_date",
            "label": _("التاريخ"),
            "fieldtype": "Date",
            "width": 100
        },
        {
            "fieldname": "posting_time",
            "label": _("الوقت"),
            "fieldtype": "Time",
            "width": 80
        },
        {
            "fieldname": "voucher_type",
            "label": _("نوع المستند"),
            "fieldtype": "Data",
            "width": 120
        },
        {
            "fieldname": "voucher_no",
            "label": _("رقم المستند"),
            "fieldtype": "Link",
            "options": "Sales Invoice",
            "width": 140
        },
        {
            "fieldname": "customer_name",
            "label": _("Customer"),
            "fieldtype": "Data",
            "width": 180
        },
        {
            "fieldname": "invoice_status",
            "label": _("حالة الفاتورة"),
            "fieldtype": "Data",
            "width": 120
        },
        {
            "fieldname": "grand_total",
            "label": _("Amount"),
            "fieldtype": "Currency",
            "width": 120
        },
        {
            "fieldname": "owner",
            "label": _("المستخدم"),
            "fieldtype": "Data",
            "width": 120
        }
    ]

def get_sales_data(filters):
    """Get sales data based on filters and user permissions"""
    conditions = []
    values = {
        "company": filters.get("company"),
        "from_date": filters.get("from_date"),
        "to_date": filters.get("to_date")
    }
    
    conditions.append("si.company = %(company)s")
    conditions.append("si.posting_date BETWEEN %(from_date)s AND %(to_date)s")
    
    # Filter explicitly by docstatus to ensure only submitted invoices are shown
    conditions.append("si.docstatus = 1")
    
    # Add time filter conditions
    if filters.get("from_time") and filters.get("to_time"):
        conditions.append("""
            CONCAT(si.posting_date, ' ', si.posting_time) BETWEEN 
            CONCAT(%(from_date)s, ' ', %(from_time)s) AND 
            CONCAT(%(to_date)s, ' ', %(to_time)s)
        """)
        values["from_time"] = filters.get("from_time")
        values["to_time"] = filters.get("to_time")
    
    # Filter by customer if provided
    if filters.get("customer"):
        conditions.append("si.customer = %(customer)s")
        values["customer"] = filters.get("customer")
    
    # Filter by invoice status (if provided)
    if filters.get("status") and filters.get("status") not in ['Draft', 'Cancelled']:
        conditions.append("si.status = %(status)s")
        values["status"] = filters.get("status")
    
    # Filter by invoice type if provided
    if filters.get("invoice_type"):
        if filters.get("invoice_type") == "Sales":
            conditions.append("si.is_return = 0")
        elif filters.get("invoice_type") == "Return":
            conditions.append("si.is_return = 1")
    
    # Apply permissions - Only show invoices the user has access to
    user = frappe.session.user
    
    # Get user roles the safer way
    user_roles = frappe.get_roles(user)
    
    # If not Administrator, System Manager, or Accounts Manager, apply user-specific filters
    if "Administrator" not in user_roles and "System Manager" not in user_roles and "Accounts Manager" not in user_roles:
        # Sales managers can see all sales invoices
        if "Sales Manager" not in user_roles:
            # Sales Users/Sales Reps can only see invoices they created
            if "Sales User" in user_roles:
                conditions.append("si.owner = %(current_user)s")
                values["current_user"] = user
            
            # If configured, also filter by sales person territory
            try:
                sales_person = frappe.db.get_value("Sales Person", {"user": user}, "name")
                if sales_person:
                    territories = frappe.db.sql("""
                        SELECT territory 
                        FROM `tabSales Person Territory` 
                        WHERE parent = %s
                    """, (sales_person,), as_dict=1)
                
                    if territories:
                        territory_list = [t.territory for t in territories]
                        territory_conditions = []
                        
                        for i, territory in enumerate(territory_list):
                            territory_param = f"territory_{i}"
                            territory_conditions.append(f"si.territory = %({territory_param})s")
                            values[territory_param] = territory
                        
                        if territory_conditions:
                            conditions.append(f"({' OR '.join(territory_conditions)})")
            except Exception as e:
                frappe.log_error(f"Error in territory filtering: {str(e)}", "Simple Sales Log")
    
    # Query to get sales invoice data
    sales_data = frappe.db.sql("""
        SELECT
            si.posting_date,
            si.posting_time,
            'Sales Invoice' as voucher_type,
            si.name as voucher_no,
            si.customer_name,
            si.status,
            si.is_return,
            si.grand_total,
            si.owner
        FROM
            `tabSales Invoice` si
        WHERE
            {conditions}
        ORDER BY
            si.posting_date DESC, si.posting_time DESC, si.name
    """.format(
        conditions=" AND ".join(conditions)
    ), values, as_dict=1)
    
    # Process and augment data
    data = []
    for row in sales_data:
        # Get invoice status in Arabic
        invoice_status = get_arabic_status(row.status, row.is_return)
        
        # Get user's full name
        user_fullname = frappe.db.get_value("User", row.owner, "full_name") if row.owner else ""
        
        # Format the voucher type in Arabic
        voucher_type = _("Sales Invoice") if not row.is_return else _("مرتجع مبيعات")
        
        # Replace null values with empty values or default texts
        posting_date = row.posting_date or ""
        posting_time = row.posting_time or ""
        voucher_no = row.voucher_no or ""
        customer_name = row.customer_name or _("غير محدد")
        grand_total = float(row.grand_total or 0)
        owner = user_fullname or row.owner or _("غير محدد")
        
        # Add the sales entry to the data
        data.append({
            "posting_date": posting_date,
            "posting_time": posting_time,
            "voucher_type": voucher_type,
            "voucher_no": voucher_no,
            "customer_name": customer_name,
            "invoice_status": invoice_status,
            "grand_total": grand_total,
            "owner": owner
        })
    
    return data

def get_arabic_status(status, is_return=0):
    """Convert invoice status to Arabic"""
    if is_return:
        return _("مرتجع مبيعات")
    
    status_map = {
        "Draft": _("مسودة"),
        "Submitted": _("مقدمة"),
        "Paid": _("مسددة بالكامل"),
        "Unpaid": _("غير مسددة"),
        "Partly Paid": _("مسددة جزئياً"),
        "Overdue": _("متأخرة السداد"),
        "Cancelled": _("ملغية"),
        "Credit Note Issued": _("إشعار دائن مصدر")
    }
    
    return status_map.get(status, status)